piper/leddialog.py

```diff
@@ -0,0 +1,106 @@
+
+    __gtype_name__ = "LedDialog"
+
+    stack = GtkTemplate.Child()
+    colorchooser = GtkTemplate.Child()
+    colorbutton = GtkTemplate.Child()
+    adjustment_brightness = GtkTemplate.Child()
+    adjustment_effect_rate = GtkTemplate.Child()
+
+    def __init__(self, ratbagd_led, *args, **kwargs):
+        """Instantiates a new LedDialog.
+
+        @param ratbagd_led The LED to configure, as ratbagd.RatbagdLed.
+        """
+        Gtk.Dialog.__init__(self, *args, **kwargs)
+        self.init_template()
+        self._led = ratbagd_led
+
+        mode = self._led.mode
+        if mode == RatbagdLed.MODE_ON:
+            self.stack.set_visible_child_name("solid")
+        elif mode == RatbagdLed.MODE_CYCLE:
+            self.stack.set_visible_child_name("cycle")
+        elif mode == RatbagdLed.MODE_BREATHING:
+            self.stack.set_visible_child_name("breathing")
+        elif mode == RatbagdLed.MODE_OFF:
+            self.stack.set_visible_child_name("off")
+        rgba = self._get_led_color_as_rgba()
+        self.colorchooser.set_rgba(rgba)
+        self.colorbutton.set_rgba(rgba)
+        self.adjustment_brightness.set_value(self._led.brightness)
+        self.adjustment_effect_rate.set_value(self._led.effect_rate)
+
+    @GtkTemplate.Callback
+    def _on_color_changed(self, chooser, pspec):
+        # The color chooser's color changed, adjust the LED's color accordingly.
+        self._set_led_color_as_rgba(chooser.get_rgba())
+
+    @GtkTemplate.Callback
+    def _on_color_set(self, button):
+        # The color button made a selection, adjust the LED's color accordingly.
+        self._set_led_color_as_rgba(button.get_rgba())
+
+    @GtkTemplate.Callback
+    def _on_brightness_changed(self, adjustment):
+        # The brightness slider changed, adjust the LED's brightness
+        # accordingly.
+        self._led.brightness = adjustment.get_value()
+
+    @GtkTemplate.Callback
+    def _on_effect_rate_changed(self, adjustment):
+        # The effect rate slider changed, adjust the LED's effect rate
+        # accordingly.
+        self._led.effect_rate = adjustment.get_value()
+
+    @GtkTemplate.Callback
+    def _on_stack_page_changed(self, stack, pspec):
+        # The stack page changed; adjust the LED's mode accordingly.
+        visible_child = stack.get_visible_child_name()
+        if visible_child == "solid":
+            self._led.mode = RatbagdLed.MODE_ON
+        elif visible_child == "cycle":
+            self._led.mode = RatbagdLed.MODE_CYCLE
+        elif visible_child == "breathing":
+            self._led.mode = RatbagdLed.MODE_BREATHING
+        elif visible_child == "off":
+            self._led.mode = RatbagdLed.MODE_OFF
+
+    def _get_led_color_as_rgba(self):
+        # Helper function to convert ratbagd's 0-255 color range to a Gdk.RGBA
+        # with a 0.0-1.0 color range.
+        r, g, b = self._led.color
+        return Gdk.RGBA(r / 255, g / 255, b / 255, 1.0)
```

Comment:
This is me having been bitten by C and Python2 too many times: use `255.0` here please to make it obvious that it's not an integer division. I know it's not in Py3, but still...

Also, we don't usually do spaces around /, afair